Src/Sarif.PatternMatcher.Sdk/ExtensionMethods.cs

```diff
@@ -1,10 +1,42 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Net.Http.Headers;
+
 namespace Microsoft.CodeAnalysis.Sarif.PatternMatcher.Sdk
 {
     public static class ExtensionMethods
     {
+        public static HttpResponseMessage ResponseHeadersRead(this HttpClient httpClient, HttpRequestMessage request)
+        {
+            return httpClient
+                .SendAsync(request, HttpCompletionOption.ResponseHeadersRead)
+                .GetAwaiter()
+                .GetResult();
+        }
+
+        public static void Merge(this HttpRequestHeaders httpRequestHeaders,
```

Comment:
Maybe change the name to be Merge Headers to be more specific? As the request might have content, which need each rule class to add separately.

Reply:
This is an extension method for `HttpRequestHeaders`, its caller we already know its a http header.
E.g. `request.Headers.Merge(...)`

Reply:
But this code doesn't merge headers into the request headers instance, it merges any collection of key value pairs.

Reply:
Utilization looks like this and I think it's pretty clear:

~~~
myHttpRequestHeaders.Merge(myDictionaryofValues);
~~~